ansible/configs/sap-hana/files/hosts_template.j2

```diff
@@ -32,4 +33,41 @@ ansible_user={{remote_user}}
 ansible_ssh_common_args="-o StrictHostKeyChecking=no"
 sap_preconfigure_modify_etc_hosts=true
 sap_domain=automation.local
+{% endif %}
+
+{% if cloud_provider == "ec2" %}
+[hanas]
+{% if instances[3].count == 2 %}
+hana1
+hana2
+{% else %}
+hana1
```

Comment:
Why don't you set `hana_instance_count` and `s4hana_instance_count` for ec2 too, so your host template would be simpler ?
You shouldn't  need to duplicate the whole block for ec2.

Reply:
Even if I add `hana_instance_count` and `s4hana_instance_count` the hostnames for OSP and AWS are different so I still need to keep 2 versions

Reply:
it's very weird to me that you don't loop over instances, get the name, then depending on the count create the entries.
Something like:
```jinja2

{% for instance in instances %}
[{{ instance.name }}s]
{% for c in range(1,(instance['count'] |int)+1) %}
{{instance['name']}}{{loop.index}}
{% endfor %}
{% endfor %}
```